omod/src/main/java/org/openmrs/module/oauth2login/web/HttpUtils.java

```diff
@@ -0,0 +1,55 @@
+package org.openmrs.module.oauth2login.web;
+
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.openmrs.api.APIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains http utility methods
+ */
+public class HttpUtils {
+	
+	protected static final Logger log = LoggerFactory.getLogger(HttpUtils.class);
+	
+	/**
+	 * Fetches JSON web keys from the Identity provider at the specified URL
+	 * 
+	 * @param url the URL of the identity provider
+	 * @return JSON web keys
+	 * @throws Exception
+	 */
+	public static String getJsonWebKeys(String url) throws Exception {
+		HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+		
+		try {
+			connection.setRequestProperty("Accept", "application/json");
+			connection.setDoInput(true);
+			connection.setUseCaches(false);
+			
+			if (log.isDebugEnabled()) {
+				log.debug("Fetching JSON web keys from identity provider");
+			}
+			
+			connection.connect();
+			
+			if (connection.getResponseCode() != 200) {
+				final String error = connection.getResponseCode() + " " + connection.getResponseMessage();
+				throw new APIException("Unexpected response " + error + " from identity provider");
+			}
+			
+			return IOUtils.toString((InputStream) connection.getContent(), StandardCharsets.UTF_8);
```

Comment:
getContent does extra stuff like applying any registered content handler but I guess in this case it has no extra effect since we don't need that, so to avoid the unnecessary typecast I guess it's better to use getInputStream